src/Workspaces/Core/Portable/Workspace/Host/TemporaryStorage/ITemporaryStorageService2.cs

```diff
@@ -13,11 +13,11 @@ internal interface ITemporaryStorageService2 : ITemporaryStorageService
         /// <summary>
         /// Attach to existing <see cref="ITemporaryStreamStorage"/> with given name.
         /// </summary>
-        ITemporaryStreamStorage AttachTemporaryStreamStorage(string storageName, long size, CancellationToken cancellationToken = default(CancellationToken));
+        ITemporaryStreamStorage AttachTemporaryStreamStorage(string storageName, long offset, long size, CancellationToken cancellationToken = default(CancellationToken));
```

Comment:
I admit I wish this API had a "temporary stream handle" struct that was name + offset + size, and that got passed around, rather than leaving it up to callers to not screw up offset/size ordering. ITemporaryStorageWithName then would become ITemporaryStorageWithHandle, or perhaps just a GetHandle call...